### DISCUSSION TITLE
Is it possible to trigger a warning when dependencies are not overridden during tests?

### DISCUSSION THREAD

**Discussion by vitorvarela** [hmlongco/Factory#221] [Q&A]
I want to ensure that we are overriding dependencies during unit tests. This is a good approach to make sure we have full control over our dependencies.
I noticed that is possible to create scopes for previews and unit tests. 
```swift
extension Container {
    var modelData: Factory<ModelData> {
        self { ModelData() }
            .scope(.session)
            .onPreview { ModelData(games: TestData.games3) }
            .onTest { ModelData(games: TestData.games4) }
            .onDebug { ModelData(games: TestData.games2) }
    }
}
```

So i would image if I did something like this: 
```
extension Container {
    var modelData: Factory<ModelData> {
        self { ModelData() }
            .scope(.session)
            .onTest { faltaErrror() }
    }
}
```

It would result in an error during the tests, causing a crash.
My question is why when I override the dependencies during tests doesn’t take precedence over `.onTest { faltaError() }`.
```swift
extension Container {
    func setupMocks() {
        modelData.register { MockData() }
    }
}
```

Is there another way to ensure we always override dependencies during unit tests, perhaps by triggering an error or warning?

**Comment by hmlongco** [comment]
If you read the documentation on context precedence you'll see why.

https://hmlongco.github.io/Factory/documentation/factory/contexts

**Reply by vitorvarela** [reply]
Yes, I read the documentation the original problem was that I wanted to ensure that we are overriding dependencies during unit tests.
Using contexts first can make it easy to forget that you need to override a dependency during tests, as you have to explicitly call the test context. Additionally, even when I use the test context, I’m no longer able to override dependencies during my tests. That's why I added "Is there another way to ensure we always override dependencies during unit tests, perhaps by triggering an error or warning when we forgot to override?"

**Reply by vitorvarela** [reply]
Hi @hmlongco, just following up on this. Do you have any advice on how to avoid relying on developers to manually override dependencies for unit tests and prevent potential issues if they forget and use the live dependency by mistake?
